.github/workflows/docker_build.yaml

```diff
@@ -0,0 +1,46 @@
+name: Build and Push Docker image
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: mbari/mbari_wec
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
```

Comment:
I think we said we only want to have 2 tags, `latest` and `previous`, to make maintenance easier.

For the initial test, this should probably just push to a test tag, so that it doesn't overwrite the working image we manually pushed.

Reply:
Handled in https://github.com/osrf/mbari_wec/pull/55/commits/a701fd17ace2f8622b0d494733a101928aaa456b, the dockerhub tags would be the git tags for this repo and a `latest-rc` tag would also be created by default.